File: xxl-ai-ui/src/modules/business/supplier/types/supplier.ts
/**
 * 供应商 类型定义
 */
import type { ListQuery } from '@/types'

/** 供应商实体（对应后端 Supplier / SupplierDTO） */
export interface Supplier {
  id?: number
  spaceId?: number
  /** 供应商名称 */
  name?: string
  /** 接口地址 */
  baseUrl?: string
  /** API密钥 */
  apiKey?: string
  /** 状态：0-正常、1-停用 */
  status?: number
  remark?: string
  addTime?: string
  updateTime?: string
  [key: string]: unknown
}

/** 搜索栏表单查询参数 */
export interface SupplierQuery {
  pageNum: number
  pageSize: number
  name?: string
  status: number
}

/** 列表接口请求参数 */
export type SupplierListQuery = ListQuery<SupplierQuery>

/** 连通测试结果（对应后端 SupplierConnectDTO） */
export interface SupplierConnect {
  /** 是否连通 */
  connectable: boolean
  /** 最近一次请求 HTTP 状态码（0 表示网络异常未收到响应） */
  httpCode: number
  /** 测试耗时（毫秒） */
  elapsedMs: number
  /** 测试过程描述 */
  message?: string
  [key: string]: unknown
}

File: xxl-ai-api/src/main/java/com/xxl/ai/api/business/supplier/model/entity/Supplier.java
package com.xxl.ai.api.business.supplier.model.entity;

import java.util.Date;

/**
 * 供应商 实体
 *
 * @author xxl-ai 2026-09-05
 */
public class Supplier {

    private long id;            /* 供应商ID */
    private long spaceId;       /* 空间ID */
    private String name;        /* 供应商名称 */
    private String baseUrl;     /* 接口地址 */
    private String apiKey;      /* API密钥 */
    private int status;         /* 状态：0-正常、1-停用 */
    private String remark;      /* 备注 */
    private Date addTime;       /* 新增时间 */
    private Date updateTime;    /* 更新时间 */

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public long getSpaceId() {
        return spaceId;
    }

    public void setSpaceId(long spaceId) {
        this.spaceId = spaceId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBaseUrl() {
        return baseUrl;
    }

    public void setBaseUrl(String baseUrl) {
        this.baseUrl = baseUrl;
    }

    public String getApiKey() {
        return apiKey;
    }

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public String getRemark() {
        return remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

    public Date getAddTime() {
        return addTime;
    }

    public void setAddTime(Date addTime) {
        this.addTime = addTime;
    }

    public Date getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(Date updateTime) {
        this.updateTime = updateTime;
    }

}